uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveBalanceChange.php

```diff
@@ -405,7 +405,7 @@ public static function getTotalBalanceChangeForLeaveRequest(LeaveRequest $leaveR
       }
     }
 
-    return $balance;
+    return abs($balance);
```

Comment:
I think some tests usually require this function to return a negative balance and not positive.
And really the result should be negative because the leave requests cause a negative balance, if you need it in positive it should be in the function where this is returned.

Reply:
`getLeaveRequestBalance` returns `calculateBalanceChangeFromCreateParams()` or `getTotalBalanceChangeForLeaveRequest()`.

`calculateBalanceChangeFromCreateParams` returns **abs()**
but 
`getTotalBalanceChangeForLeaveRequest` doesn't.

Is that expected?

Reply:
Yeah its expected. If you check the code `calculateBalanceChangeFromCreateParams` returns an absolute value for the results returned by the `LeaveRequest::calculateBalanceChange` function, You need to wrap the value returned from  `getTotalBalanceChangeForLeaveRequest` in `abs` also as we need the leave balance in positive to compare it against the current entitlement balance.

Reply:
Yep, did exactly what you said ^^. Was not sure that's the best solution, but since you suggest the same, I am not much more confident. Thank you.